runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/source/unbounded/FlinkUnboundedSourceReader.java

```diff
@@ -103,22 +104,30 @@ public void start() {
     createPendingBytesGauge(context);
     Long watermarkInterval =
         pipelineOptions.as(FlinkPipelineOptions.class).getAutoWatermarkInterval();
-    if (watermarkInterval != null) {
-      scheduleTaskAtFixedRate(
-          () -> {
-            // Set the watermark emission flag first.
-            shouldEmitWatermark = true;
-            // Wake up the main thread if necessary.
-            CompletableFuture<Void> f = dataAvailableFutureRef.get();
-            if (f != DUMMY_FUTURE) {
-              f.complete(null);
-            }
-          },
-          watermarkInterval,
-          watermarkInterval);
-    } else {
-      LOG.warn("AutoWatermarkInterval is not set, watermarks won't be emitted.");
+    if (watermarkInterval == null) {
+      watermarkInterval =
+          (pipelineOptions.as(FlinkPipelineOptions.class).getMaxBundleTimeMills()) / 5L;
+      watermarkInterval =
+          (watermarkInterval > MIN_WATERMARK_EMIT_INTERVAL_MS)
+              ? watermarkInterval
+              : MIN_WATERMARK_EMIT_INTERVAL_MS;
+      LOG.warn(
+          "AutoWatermarkInterval is not set, watermarks will be emitted at a default interval of "
+              + watermarkInterval
+              + "ms");
```

Comment:
```suggestion
      LOG.warn(
          "AutoWatermarkInterval is not set, watermarks will be emitted at a default interval of {} ms",
            watermarkInterval);
```

Reply:
This should be the default way of logging using slf4j, it is generally preferred over string concatenation. Please commit the change and squash the commits again, then ping me back, I'll merge that. Thanks for the contribution!